tests/channel_internal_ut/channel_internal_ut.c

```diff
@@ -67,57 +78,72 @@ static void test_data_dispose(void* data)
     (void)data;
 }
 
-static void test_pull_callback_abandoned(void* context, CHANNEL_CALLBACK_RESULT result, THANDLE(RC_PTR) data)
+static void test_pull_callback_abandoned(void* context, CHANNEL_CALLBACK_RESULT result, THANDLE(RC_STRING) pull_correlation_id, THANDLE(RC_STRING) push_correlation_id, THANDLE(RC_PTR) data)
 {
     ASSERT_ARE_EQUAL(int, 0, (*(int32_t*)context)++);
     ASSERT_ARE_EQUAL(CHANNEL_CALLBACK_RESULT, CHANNEL_CALLBACK_RESULT_ABANDONED, result);
+    ASSERT_ARE_EQUAL(void_ptr, g.g_correlation_id, pull_correlation_id);
+    ASSERT_IS_NULL(push_correlation_id);
     ASSERT_IS_NULL(data);
 }
 
-static void test_push_callback_abandoned(void* context, CHANNEL_CALLBACK_RESULT result)
+static void test_push_callback_abandoned(void* context, CHANNEL_CALLBACK_RESULT result, THANDLE(RC_STRING) pull_correlation_id, THANDLE(RC_STRING) push_correlation_id)
 {
     ASSERT_ARE_EQUAL(int, 0, (*(int32_t*)context)++);
     ASSERT_ARE_EQUAL(CHANNEL_CALLBACK_RESULT, CHANNEL_CALLBACK_RESULT_ABANDONED, result);
+    ASSERT_IS_NULL(pull_correlation_id);
+    ASSERT_ARE_EQUAL(void_ptr, g.g_correlation_id, push_correlation_id);
 }
 
-static void test_pull_callback_ok(void* context, CHANNEL_CALLBACK_RESULT result, THANDLE(RC_PTR) data)
+static void test_pull_callback_ok(void* context, CHANNEL_CALLBACK_RESULT result, THANDLE(RC_STRING) pull_correlation_id, THANDLE(RC_STRING) push_correlation_id, THANDLE(RC_PTR) data)
 {
     ASSERT_ARE_EQUAL(int, 0, (*(int32_t*)context)++);
```

Comment:
>++ [](http://example.com/codeflow?start=49&length=2)

++ should be postponed to after the ASSERT_ARE_EQUAL due to evaluation order and as it is written is relies on implementation details of "ASSERT_ARE_EQUAL" which can change at any time.

Here and everywhere.

Reply:
Fixed.